CHEWBBACA/CHEWBBACA_NS/Extract_cgAlleles.py

```diff
@@ -0,0 +1,155 @@
+
+        return (np.sum(column) / column_length) >= cgPercent
+
+    # remove genomes
+    for genome in calls.index:
+
+        if genome in listgenomesRemove:
+
+            print("removed genome : {}".format(genome))
+
+    to_remove_bool = calls.index.isin(listgenomesRemove)
+
+    calls_removed = calls.loc[~ to_remove_bool]
+
+    print ("building the presence and absence matrix...")
+
+    presence_absence = calls_removed.apply(binarize)
+
+
+    pa_path = os.path.join(outputfile, 'Presence_Absence.tsv')
+    presence_absence.to_csv(pa_path, sep='\t')
+
+    print ("presence and abscence matrix built")
+
+    pa_rows, _ = presence_absence.shape
+    is_above_threshold = presence_absence.apply(above_threshold,
+                                                args=(pa_rows,))
+
+    columns_to_remove = calls_removed.columns[~ is_above_threshold]
+
+    return presence_absence, calls_removed, columns_to_remove
+
+def missing_data_table(cleaned_calls):
+
+    n_genes = len(cleaned_calls.columns)
+    genes_present = cleaned_calls.apply(np.count_nonzero, axis=1)
+
+    missing_data = {'FILE': cleaned_calls.index,
+                    'number of missing data': n_genes - genes_present,
+                    'percentage': 1 - (genes_present / n_genes)}
+
+    missing_data_df = pd.DataFrame(missing_data,
+                                   columns=['FILE',
+                                            'number of missing data',
+                                            'percentage'])
+
+    return missing_data_df
+
+
+def clean(inputfile, outputfile, toremovegenes, toremovegenomes, cgPercent):
+
+    # open the raw file to be clean
+    calls = pd.read_csv(inputfile,
+                        header=0, index_col=0, sep='\t', low_memory=False)
+
+    # get presence abscence matrix
+    presence_absence, calls_removed, cols_to_remove = presAbs(calls,
+                                                              toremovegenomes,
+                                                              outputfile,
+                                                              cgPercent)
+
+
+    total_genes_to_delete = set(toremovegenes).union(set(cols_to_remove))
+
+    # clean the original matrix,
+    # using the information on the presence/abscence matrix
+    print ("processing the matrix")
+
+    to_keep = ~calls_removed.columns.isin(total_genes_to_delete)
+
+    cleaned = calls_removed.loc[:, to_keep]
+
+    # remove INF and other missing data tags from the profile
+    words = ['LNF', 'PLOT3', 'PLOT5', 'ASM', 'ALM', 'NIPHEM', 'NIPH', 'LOTSC']
+    for word in words:
+        cleaned = cleaned.replace(to_replace='{}.*'.format(word),
+                                  value='0', regex=True)
+
+    #need to replace the * alleles for a high number, simple remove the * will make alleles to be repeated while not being the same
+    #iterate through columns and get uniques
+
+    for column in cleaned:
+        maxAllele=999999
```

Comment:
why do we need a maxAllele?

Reply:
It seems that the initial version of SyncSchema would allow assignment of the same integer identifier for different alleles. One could have * and the other would not. Right now, each integer identifier can only be assigned to one sequence. We'll remove this part so that it works as intended.